jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TraceContextCodecTest.java

```diff
@@ -160,10 +164,31 @@ public void testEmptyTraceStateNotPropagated() {
     assertEquals(EXAMPLE_TRACE_PARENT, injectCarrier.get(TRACE_PARENT));
   }
 
+  @Test
+  public void testDebugIdWithoutTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertEquals(EXAMPLE_DEBUG_ID, spanContext.getDebugId());
+  }
+
+  @Test
+  public void testDebugIdWithTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(TRACE_PARENT, EXAMPLE_TRACE_PARENT);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertEquals(EXAMPLE_DEBUG_ID, spanContext.getDebugId());
```

Comment:
Technically, this test is identical to the previous one in terms of asserts. It's worth verifying that the TRACE_PARENT was also parsed correctly, e.g. by asserting the trace ID.